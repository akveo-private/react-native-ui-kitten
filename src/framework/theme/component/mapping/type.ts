// TODO(mapping/type): declare Config/Token type

export type ThemeMappingConfigType = any;

export interface ThemeMappingType {
  parameters: string[];
  variants: any;
}

export interface VariantType {
  state: any;
}

export type TokenType = any;
