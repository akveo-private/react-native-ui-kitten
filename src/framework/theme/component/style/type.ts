export enum Interaction {
  ACTIVE = 'active',
}

export enum State {
  CHECKED = 'checked',
  DISABLED = 'disabled',
  FOCUSED = 'focused',
}

export namespace Interaction {
  export function parse(interaction: string): Interaction | undefined {
    return Interaction[interaction.toUpperCase()];
  }
}

export namespace State {
  export function parse(state: string): State | undefined {
    return State[state.toUpperCase()];
  }
}
