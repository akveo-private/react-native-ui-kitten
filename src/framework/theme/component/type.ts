// TODO(@type): declare theme types

export type ThemeType = any;
export type ThemedStyleType = any;
export type StyleSheetType = any;
