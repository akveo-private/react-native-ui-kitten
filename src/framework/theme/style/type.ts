export type StyleType = any;

export enum Interaction {
  ACTIVE = 'active',
  FOCUSED = 'focused',
  INDETERMINATE = 'indeterminate',
}

export enum State {
  CHECKED = 'checked',
  SELECTED = 'selected',
  DISABLED = 'disabled',
}
