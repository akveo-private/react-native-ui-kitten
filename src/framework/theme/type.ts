export enum Interaction {
  ACTIVE = 'active',
}

export enum State {
  CHECKED = 'checked',
  SELECTED = 'selected',
  DISABLED = 'disabled',
  FOCUSED = 'focused',
}

export namespace Interaction {
  export function parse(interaction: string): Interaction | undefined {
    return Interaction[interaction.toUpperCase()];
  }
}

export namespace State {
  export function parse(state: string): State | undefined {
    return State[state.toUpperCase()];
  }
}

export type ThemeType = any;
export type StyleType = any;
export type ThemedStyleType = any;
export type StyleSheetType = any;

export interface ModalComponentCloseProps {
  onCloseModal?: () => void;
}
