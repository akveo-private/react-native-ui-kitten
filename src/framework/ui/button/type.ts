export interface ButtonAlignment {
  rawValue: string;

  flex(): FlexAlignment;
}

export type FlexAlignment = 'row' | 'row-reverse';

export class ButtonAlignments {
  static LEFT: ButtonAlignment = new class implements ButtonAlignment {
    rawValue: string = 'left';

    flex(): FlexAlignment {
      return 'row';
    }
  };

  static RIGHT: ButtonAlignment = new class implements ButtonAlignment {
    rawValue: string = 'right';

    flex(): FlexAlignment {
      return 'row-reverse';
    }
  };

  static parse(value: string | ButtonAlignment, fallback?: ButtonAlignment): ButtonAlignment | undefined {
    if (ButtonAlignments.typeOf(value)) {
      return value;
    }

    return ButtonAlignments.parseString(value, fallback);
  }

  private static parseString(rawValue: string, fallback?: ButtonAlignment): ButtonAlignment | undefined {
    switch (rawValue) {
      case ButtonAlignments.LEFT.rawValue:
        return ButtonAlignments.LEFT;
      case ButtonAlignments.RIGHT.rawValue:
        return ButtonAlignments.RIGHT;
      default:
        return fallback;
    }
  }

  private static typeOf(value: any): value is ButtonAlignment {
    const { rawValue } = (<ButtonAlignment>value);

    return rawValue !== undefined;
  }
}
