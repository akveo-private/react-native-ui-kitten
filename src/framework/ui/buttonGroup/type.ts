import { StyleType } from '@kitten/theme';

export interface ButtonStyleProvider {
  start(source: StyleType): StyleType;

  center(source: StyleType): StyleType;

  end(source: StyleType): StyleType;
}

export class ButtonStyleProviders {
  static DEFAULT: ButtonStyleProvider = new class implements ButtonStyleProvider {

    defaults(source: StyleType): StyleType {
      return {
        borderRadius: 0,
      };
    }

    start(source: StyleType): StyleType {
      const { borderRadius, margin } = source;

      return {
        ...this.defaults(source),
        borderTopStartRadius: borderRadius,
        borderBottomStartRadius: borderRadius,
        marginRight: margin / 2,
      };
    }

    center(source: StyleType): StyleType {
      const { margin } = source;

      return {
        ...this.defaults(source),
        marginHorizontal: margin / 2,
      };
    }

    end(source: StyleType): StyleType {
      const { borderRadius, margin } = source;

      return {
        ...this.defaults(source),
        borderTopEndRadius: borderRadius,
        borderBottomEndRadius: borderRadius,
        marginLeft: margin / 2,
      };
    }
  };
}
