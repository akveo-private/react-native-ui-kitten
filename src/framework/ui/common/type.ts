import {
  NativeSyntheticEvent,
  NativeScrollEvent,
  TextInputFocusEventData,
  TextInputEndEditingEventData,
  TouchableOpacityProps,
  GestureResponderEvent,
} from 'react-native';

export type ScrollEvent = NativeSyntheticEvent<NativeScrollEvent>;
export type InputFocusEvent = NativeSyntheticEvent<TextInputFocusEventData>;
export type InputEndEditEvent = NativeSyntheticEvent<TextInputEndEditingEventData>;

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

// @ts-ignore: props override
export interface TouchableOpacityIndexedProps extends TouchableOpacityProps {
  onPress?: (event: GestureResponderEvent, index: number) => void;
  onPressIn?: (event: GestureResponderEvent, index: number) => void;
  onPressOut?: (event: GestureResponderEvent, index: number) => void;
  onLongPress?: (event: GestureResponderEvent, index: number) => void;
}
