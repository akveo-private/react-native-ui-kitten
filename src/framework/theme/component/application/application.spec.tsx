import React from 'react';
import { render } from 'react-native-testing-library';
import {
  View,
  Text,
} from 'react-native';
import {
  ApplicationProvider,
  Props as ApplicationProps,
} from './applicationProvider.component';
import {
  mapping,
  theme,
} from './application.spec.config';
import {
  ThemeMappingType,
  ThemeMapType,
} from 'eva/packages/common';
import { ThemeType } from '../../type';

describe('@app: application wrapper check', () => {

  interface ApplicationState {
    mapping: ThemeMappingType;
    styles: ThemeMapType;
    theme: ThemeType;
  }

  class TestApplicationWrapper extends React.Component<ApplicationProps, ApplicationState> {

    constructor(props) {
      super(props);
      this.state = {
        mapping: props.mapping,
        styles: props.style,
        theme: props.theme,
      };
    }

    render() {
     return (
       <ApplicationProvider
         mapping={this.state.mapping}
         styles={this.state.styles}
         theme={this.state.theme}>
         <View><Text>Application Provider</Text></View>
       </ApplicationProvider>
     );
    }
  }

  it('* renders properly', () => {
    const component = render(
      <TestApplicationWrapper mapping={mapping} styles={{}} theme={theme}>
        <View><Text>Test</Text></View>
      </TestApplicationWrapper>,
    );

    expect(component).toMatchSnapshot();
  });

});
