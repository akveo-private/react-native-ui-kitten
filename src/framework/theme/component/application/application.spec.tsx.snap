// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@app: application wrapper check * renders properly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text>
      Application Provider
    </Text>
  </View>
</View>
`;
